gen/templates/generic/data_source.go

```diff
@@ -152,9 +160,25 @@ func (d *{{camelCase .Name}}DataSource) Read(ctx context.Context, req datasource
 		return
 	}
 
-	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
 
-	res, err := d.client.Get("{{if .GetRestEndpoint}}{{.GetRestEndpoint}}{{else}}{{.RestEndpoint}}{{end}}" + config.Id.ValueString())
+	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
+	{{if .RemoveId}}
+
+
+		var params = "?"
+		{{range $i, $a := .Attributes}}
+		{{- if not .Value}}
+		{{- if .QueryParam}}
+		if(!config.{{toGoName .TfName}}.IsNull()) {
+			params = params + "{{.ModelName}}=" + url.QueryEscape(config.{{toGoName .TfName}}.ValueString() )+ "&"
```

Comment:
Is the extra "&" at the end, syntactically correct?

Reply:
Yea that should have no impact. Otherwise your needing to count indexes and whether its first or last. In terms of the api call, would just have an extra & on the end. No issue